Configuration/PyReleaseValidation/python/MatrixInjector.py

```diff
@@ -356,6 +358,16 @@ def prepare(self,mReader, directories, mode='init'):
                                     ns=map(int,arg[arg.index('--relval')+1].split(','))
                                     chainDict['nowmTasklist'][-1]['RequestNumEvents'] = ns[0]
                                     chainDict['nowmTasklist'][-1]['EventsPerJob'] = ns[1]
+                                    chainDict['nowmTasklist'][-1]['EventsPerLumi'] = ns[1]
+                                    #overwrite EventsPerLumi if numberEventsInLuminosityBlock is set in cmsDriver
+                                    if 'numberEventsInLuminosityBlock' in s[2][index]:
+                                        nEventsInLuminosityBlock = (((s[2][index].replace(" ","").replace("numberEventsInLuminosityBlock=cms.untracked.uint32("," ")).split()[1]).split(')'))[0]
```

Comment:
I'm worried that this is a somewhat fragile construct. For example, if `numberEventsInLuminosityBlock` is set on one line with the type `cms.untracked.uint32()`, and then changed on a subsequent line (when specifying the type is not necessary), this won't pick up the final value.

Also, this currently returns a string; should it be converted to an int?

Proposal:
```
nEventsInLuminosityBlock = eval(s[2][index].split("=")[-1])
if isinstance(nEventsInLuminosityBlock,cms.uint32): nEventsInLuminosityBlock = nEventsInLuminosityBlock.value()
```

This gets the RHS of the expression as its actual type (and then gets the value if it's a cms type). Still not foolproof, would be better to `exec()` the whole config and extract the value, but maybe good enough as a starting point.

Reply:
@kpedro88 
Thanks. Maybe the first question is do we need this first? For example, if it's useful to split all GEN relvals anyways, having the default parameter set, it should be enough. I think this setting is there, just to make LS crossing in IB. We don't set it in any regular relvals.

Reply:
@srimanob I think the code in my proposal is a bit simpler/easier to parse (eval + only one split vs. two replaces, two splits, multiple indices). Also, I foresee some relvals testing RandomizedParameters eventually, which would also require this parameter, and it may not be provided in exactly the same way.

Reply:
Hi,
Oh, I mean I can change, I have no objection in code improvement. When I say "do we need this first?", I mean the setting using this method, instead of using parameter defined in runTheMatrix directly, also in this PR.

Reply:
Why even use a CMS specific type at all? The parameter is never used by the C++ code, correct?

Reply:
@Dr15Jones I'm not sure what you mean; `numberEventsInLuminosityBlock` is definitely used in the C++ code: https://github.com/cms-sw/cmssw/search?l=C%2B%2B&q=numberEventsInLuminosityBlock

Reply:
My bad.

Reply:
Not sure how the proposal works?
s[2][index] is the full cmsDriver (*), not just
`process.source.numberEventsInLuminosityBlock=cms.untracked.uint32(5)`

Do I miss understand anything? Thx.

(*)
`cmsDriver.py TTbar_13TeV_TuneCUETP8M1_cfi --conditions auto:phase1_2018_realistic -n 10 --customise_commands "process.source.numberEventsInLuminosityBlock=cms.untracked.uint32(5)" --era Run2_2018 --relval 9000,50 -s GEN,SIM --nStreams 4 --datatier GEN-SIM --beamspot Realistic25ns13TeVEarly2018Collision --eventcontent RAWSIM --geometry DB:Extended --nConcurrentLumis 2 --io ProdTTbar_13UP18ml.io --python ProdTTbar_13UP18ml.py --no_exec --fileout file:step1.root`

Reply:
Ah, I misunderstood; I thought that variable contained lines from a config file, not a cmsDriver command. In that case:
```
import shlex
lumiCmds = [cmd for cmd in shlex.split(s[2][index]) if 'numberEventsInLuminosityBlock' in cmd]
if len(lumiCmds)>0:
    nEventsInLuminosityBlock = eval(lumiCmds[-1].split("=")[-1])
    if isinstance(nEventsInLuminosityBlock,cms.uint32): nEventsInLuminosityBlock = nEventsInLuminosityBlock.value()
```

Here I use `shlex.split()` to separate the command-line arguments (best practice), and then use the last argument that contains the desired variable to extract the value.

Reply:
Is `eval(lumiCmds[-1].split("=")[-1])` really work with statement like`cms.untracked.uint32(5)` ?
It seems to me, eval tries to evaluate the string `cms.untracked.uint32(5)` then lead to 
`NameError: name 'cms' is not defined`

Is it not straight forward to go with regex instead, and check condition of output? This comes without any loops.
```
ss = re.findall('process.source.numberEventsInLuminosityBlock=cms.untracked.uint32\(([ 0-9 ]*)\)', s,re.DOTALL)
if(ss[-1].isDigit):
   nEventsInLuminosityBlock = int(ss[-1])
```
As prefer, I have no strong opinion. Maybe some guide on how to make eval works with this case.

Reply:
This works fine for me in a python terminal:
```
import FWCore.ParameterSet.Config as cms
x = eval("cms.untracked.uint32(5)")
print x
```

But the regex solution also looks reasonable to me.

Reply:
Done, thx.